jpegenc/subblocks/dct_2d/dct_2d.py

```diff
@@ -0,0 +1,232 @@
+    stage_2_inst_5 = dct_1d(inputs_2nd_stage_5, outputs_2nd_stage_5, clock,
+                            reset, num_fractional_bits)
+    stage_2_inst_6 = dct_1d(inputs_2nd_stage_6, outputs_2nd_stage_6, clock,
+                            reset, num_fractional_bits)
+    stage_2_inst_7 = dct_1d(inputs_2nd_stage_7, outputs_2nd_stage_7, clock,
+                            reset, num_fractional_bits)
+
+    data_in_signed = Signal(intbv(0, min=-input_1d_stage_1.in_range,
+                                  max=input_1d_stage_1.in_range))
+    data_valid_reg = Signal(bool(0))
+    data_valid_reg2 = Signal(bool(0))
+
+    counter = Signal(intbv(0, min=0, max=8))
+
+    @always_seq(clock.posedge, reset=reset)
+    def input_subtract():
+        if inputs.data_valid:
+            data_in_signed.next = inputs.data_in
+            input_1d_stage_1.data_in.next = data_in_signed - 128
+            data_valid_reg.next = inputs.data_valid
+            input_1d_stage_1.data_valid.next = data_valid_reg
+
+    @always_seq(clock.posedge, reset=reset)
+    def first_stage_to_second():
+        inputs_2nd_stage_0.data_in.next = first_1d_output.out0
+        inputs_2nd_stage_0.data_valid.next = first_1d_output.data_valid
+        inputs_2nd_stage_1.data_in.next = first_1d_output.out1
+        inputs_2nd_stage_1.data_valid.next = first_1d_output.data_valid
+        inputs_2nd_stage_2.data_in.next = first_1d_output.out2
+        inputs_2nd_stage_2.data_valid.next = first_1d_output.data_valid
+        inputs_2nd_stage_3.data_in.next = first_1d_output.out3
+        inputs_2nd_stage_3.data_valid.next = first_1d_output.data_valid
+        inputs_2nd_stage_4.data_in.next = first_1d_output.out4
+        inputs_2nd_stage_4.data_valid.next = first_1d_output.data_valid
+        inputs_2nd_stage_5.data_in.next = first_1d_output.out5
+        inputs_2nd_stage_5.data_valid.next = first_1d_output.data_valid
+        inputs_2nd_stage_6.data_in.next = first_1d_output.out6
+        inputs_2nd_stage_6.data_valid.next = first_1d_output.data_valid
+        inputs_2nd_stage_7.data_in.next = first_1d_output.out7
+        inputs_2nd_stage_7.data_valid.next = first_1d_output.data_valid
+
+    @always_seq(clock.posedge, reset=reset)
+    def second_stage_output():
+        outputs.y00.next = outputs_2nd_stage_0.out0
+        outputs.y01.next = outputs_2nd_stage_1.out0
+        outputs.y02.next = outputs_2nd_stage_2.out0
+        outputs.y03.next = outputs_2nd_stage_3.out0
+        outputs.y04.next = outputs_2nd_stage_4.out0
+        outputs.y05.next = outputs_2nd_stage_5.out0
+        outputs.y06.next = outputs_2nd_stage_6.out0
+        outputs.y07.next = outputs_2nd_stage_7.out0
+        outputs.y10.next = outputs_2nd_stage_0.out1
+        outputs.y11.next = outputs_2nd_stage_1.out1
+        outputs.y12.next = outputs_2nd_stage_2.out1
+        outputs.y13.next = outputs_2nd_stage_3.out1
+        outputs.y14.next = outputs_2nd_stage_4.out1
```

Comment:
@cfelton is there any way to avoid these assignments with less code?
I tried setattr() but it is not convertible